pennylane_lightning/src/tests/TestingUtils.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+// Helper macro to check if an exception is thrown with the correct complete error message
+#define EXPECT_THROW_WITH_MESSAGE(stmt, etype, whatstring) EXPECT_THROW( \
+        try { \
+            stmt; \
+        } catch (const etype& ex) { \
+            EXPECT_EQ(std::string(ex.what()), whatstring); \
+            throw; \
+        } \
+    , etype)
+
+// Helper macro to check if an exception is thrown with the correct error message substring
+#define EXPECT_THROW_WITH_MESSAGE_SUBSTRING(stmt, etype, whatstring) EXPECT_THROW( \
+        try { \
+            stmt; \
+        } catch (const etype& ex) { \
+            EXPECT_NE(std::string(ex.what()).find(whatstring), std::string::npos); \
+            throw; \
+        } \
```

Comment:
What's the purpose of the backticks?

Reply:
I'm wondering the same. I would think it's just for line-continuation, but it doesn't make sense having it here, right?

Reply:
The backslashes are indeed added for line continuation because this is a macro so it gets blindly replaced to the code defined before compilation.